### PR TITLE
[Android] fix: network call timeout

### PR DESCRIPTION
**Pull request by troZee**
## Summary:

This PR is a continuation of this https://github.com/facebook/react-native/pull/29471. All context you can find here: 

https://github.com/axios/axios/issues/2073#issuecomment-1050917487 
https://github.com/axios/axios/issues/2073#issuecomment-1674817612

## Changelog:

[ANDROID] [FIXED] - change `connectTimeout` to `callTimeout` in OKHttp client

## Test Plan:

- Throttle internet connection (You can use GSM one)

```
import axios from "axios";

axios.get('https://jsonplaceholder.typicode.com/users', {
  timeout: 5000
})
.then(response => {
  console.log(response.data);
})
.catch(error => {
  if (error.code === 'ECONNABORTED') {
    console.log('Request timed out');
  } else {
    console.log(error.message);
  }
});
```

**Expected result** 

`Request timed out` should be visible in a console

**Videos** 

Before 

https://github.com/facebook/react-native/assets/12766071/03a3d666-6a27-4d67-bc68-6bcf6e908e19

After 

https://github.com/facebook/react-native/assets/12766071/2650d02d-640c-43fe-b1e8-baeac78870f2

**Example URL** 

Branch:
https://github.com/troZee/react-native/tree/fix/network-timeout-example 

Commit: 
https://github.com/troZee/react-native/commit/e255e1a423143c4bd4f3e37656a56ba2ecbbf3e2

### CODE REVIEW

**Comment by NickGerleman** [comment]
I looked through to see what is piping timeout to native, and confirmed new behavior is the expectation for `XMLHttpRequest` timeout prop (fetch just polyfills that). So, change LGTM.

**Review by NickGerleman — APPROVED** [review]

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java [hunk 1]
```diff
@@ -349,7 +349,7 @@ public void onProgress(
     // shared under the hood.
     // See https://github.com/square/okhttp/wiki/Recipes#per-call-configuration for more information
     if (timeout != mClient.connectTimeoutMillis()) {
```
Comment (NickGerleman) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java:351]: This line is no longer correct. I can fix it up quickly.

**Comment by NickGerleman** [comment]
It looks like RN is internally built using `okhttp` 3.6.0, which is way behind OSS, and doesn't yet have these APIs (they are introduced in 3.12.0). I will take a look at how painful it would be to bump this dependency in internal apps.

**Comment by troZee** [comment]
Hey 👋 
Thank you @NickGerleman for checking that issue. Could we plan a migration from 3.6.0 to 3.12.0? Could you estimate, if this migration is complex to do?  Maybe we should consider creating a separate task to bump the oss version and the internal one to https://github.com/square/okhttp/tree/parent-3.14.9 ? 

Moreover, https://square.github.io/okhttp/changelogs/changelog_3x/#version-3149 has 2 years and maybe we should consider migrating it to the newest one https://square.github.io/okhttp/changelogs/changelog_4x/#version-4100 ?

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java (hunk 1 above)
Comment (Marco29001) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java:351]: how change this line?

**Comment by hasan642** [comment]
I changed the lines you mentioned and still not working!
@NickGerleman 

is working with you? @Marco29001

**Comment by Marco29001** [comment]
I tested this with a pre-release version and if the timeout issue mentioned in this PR works, but there is still an issue that I think is related to this issue, what happens is that if the connection is lost for a long time when I recover it, it happens to me. I can't make a new request until a certain amount of time passes, this only happens on Android
@hasan642

**Comment by hasan642** [comment]
@Marco29001 we need a solution please
Axios android duplicates request to 5 times .

**Comment by troZee** [comment]
@Marco29001 @hasan642 could you provide a repro for that? 
[Here](https://github.com/troZee/react-native/tree/fix/network-timeout-example) I could not reproduce that

**Pull request closed without merging**